src/data/matrix/notation.lean

```diff
@@ -71,6 +71,26 @@ v 0
 def vec_tail {n : ℕ} (v : fin n.succ → α) : fin n → α :=
 v ∘ fin.succ
 
+variables {m n : ℕ}
+
+/-- Use `![...]` notation for displaying a vector `fin n → α`, for example:
+
+```
+#eval ![1, 2] + ![3, 4] -- ![4, 6]
+```
+-/
+instance [has_repr α] : has_repr (fin n → α) :=
+{ repr := λ f, "![" ++ (string.intercalate ", " ((list.fin_range n).map (λ n, repr (f n)))) ++ "]" }
+
+/-- Use `![...]` notation for displaying a `fin`-indexed matrix, for example:
+
+```
+#eval ![![1, 2], ![3, 4]] + ![![3, 4], ![5, 6]] -- ![![4, 6], ![8, 10]]
+```
+-/
+instance [has_repr α] : has_repr (matrix (fin m) (fin n) α) :=
+(infer_instance : has_repr (fin m → fin n → α))
```

Comment:
```suggestion
(by apply_instance : has_repr (fin m → fin n → α))
```
gives a shorter term, and gives better information in doc-gen.

Reply:
I see, is this a rule of thumb for (`Type`-valued) definitions in general?

Reply:
The problem with `infer_instance` is that if I actually want to know which instance was found, I have to `set_option pp.implicit true`, and I can't do that in doc-gen. I don't think lean itself cares other than having to unfold `infer_instance` which is easy, but I could be wrong.